x/concentrated-liquidity/incentives_test.go

```diff
@@ -3535,6 +3527,81 @@ func (s *KeeperTestSuite) TestQueryAndClaimAllIncentives() {
 	}
 }
 
+func (s *KeeperTestSuite) TestClaimAllIncentivesForPosition() {
+	testCases := []struct {
+		name             string
+		blockTimeElapsed time.Duration
+		expectedCoins    sdk.Coins
+		expectedForfeit  int
+		expectedErr      bool
+	}{
+		{
+			name:             "Claim with same blocktime",
+			blockTimeElapsed: 0,
+			expectedCoins:    sdk.NewCoins(),
+			expectedForfeit:  0,
+		},
+		{
+			name:             "Claim after 1 minute",
+			blockTimeElapsed: time.Minute,
+			expectedCoins:    sdk.NewCoins(sdk.NewCoin(USDC, sdk.NewInt(59))), //  after 1min = 59.999999999901820104usdc ~ 59usdc becasue 1usdc emitted every second
+			expectedForfeit:  0,
+		},
+		{
+			name:             "Claim after 1 hr",
+			blockTimeElapsed: time.Hour,
+			expectedCoins:    sdk.NewCoins(sdk.NewCoin(USDC, sdk.NewInt(3599))), //  after 1min = 59.999999999901820104usdc ~ 59usdc becasue 1usdc emitted every second
+			expectedForfeit:  0,
+		},
+		{
+			name:             "Claim after 3 hours",
+			blockTimeElapsed: time.Hour * 3,
+			expectedCoins:    sdk.NewCoins(sdk.NewCoin(USDC, sdk.NewInt(9999))), //  after 3hr > 2hr46min = 9999usdc.999999999901820104 ~ 9999usdc
+			expectedForfeit:  0,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			// Init suite for the test.
+			s.SetupTest()
+
+			requiredBalances := sdk.NewCoins(sdk.NewCoin(ETH, sdk.NewInt(1_000_000)), sdk.NewCoin(USDC, sdk.NewInt(5_000_000_000)))
+			s.FundAcc(s.TestAccs[0], requiredBalances)
+			s.FundAcc(s.TestAccs[1], requiredBalances)
+
+			// Create CL pool
+			pool := s.PrepareConcentratedPool()
+
+			// Set up position
+			positionIdOne, _, _, _, _, _, _, err := s.App.ConcentratedLiquidityKeeper.CreatePosition(s.Ctx, pool.GetId(), s.TestAccs[0], requiredBalances, sdk.ZeroInt(), sdk.ZeroInt(), DefaultLowerTick, DefaultUpperTick)
+			s.Require().NoError(err)
+
+			// Set incentives for pool to ensure accumulators work correctly
+			testIncentiveRecord := types.IncentiveRecord{
+				PoolId:               pool.GetId(),
+				IncentiveDenom:       USDC,
+				IncentiveCreatorAddr: s.TestAccs[1].String(), // different address is going to create incentives
+				IncentiveRecordBody: types.IncentiveRecordBody{
+					RemainingAmount: sdk.NewDec(10_000), // 2hr 46m to emit all incentives
+					EmissionRate:    sdk.NewDec(1),      // 1 per second
+					StartTime:       s.Ctx.BlockTime(),
+				},
+				MinUptime: time.Nanosecond,
+			}
+			err = s.App.ConcentratedLiquidityKeeper.SetMultipleIncentiveRecords(s.Ctx, []types.IncentiveRecord{testIncentiveRecord})
+			s.Require().NoError(err)
+
+			s.Ctx = s.Ctx.WithBlockTime(s.Ctx.BlockTime().Add(tc.blockTimeElapsed))
+
+			collectedInc, forfeitInc, err := s.App.ConcentratedLiquidityKeeper.ClaimAllIncentivesForPosition(s.Ctx, positionIdOne)
+			s.Require().NoError(err)
+			s.Require().Equal(tc.expectedCoins, collectedInc)
+			s.Require().Equal(tc.expectedForfeit, len(forfeitInc))
```

Comment:
Why do we do a length check here instead of the actual value?

Reply:
fixed